openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -474,71 +537,259 @@ OpenAssessment.ResponseView.prototype = {
+                );
+                errorCheckerTriggered = true;
+                break;
+            } else if (uploadType === "custom" && this.data.FILE_TYPE_WHITE_LIST.indexOf(ext) === -1) {
+                this.baseView.toggleActionError(
+                    'upload',
+                    gettext("You can upload files with these file types: ") +
+                    this.data.FILE_TYPE_WHITE_LIST.join(", ")
+                );
+                errorCheckerTriggered = true;
+                break;
+            } else if (this.data.FILE_EXT_BLACK_LIST.indexOf(ext) !== -1) {
+                this.baseView.toggleActionError(
+                    'upload',
+                    gettext("File type is not allowed.")
+                );
+                errorCheckerTriggered = true;
+                break;
+            }
+        }
+
+        if (!errorCheckerTriggered) {
             this.baseView.toggleActionError('upload', null);
             this.files = files;
+            this.updateFilesDescriptionsFields(files, descriptions, uploadType);
+        }
+
+        if (this.files === null) {
+            sel.find('.file__upload').prop('disabled', true);
+        }
+    },
+
+    /**
+     Render textarea fields to input description for each uploaded file.
+
+     */
+    /* jshint -W083 */
+    updateFilesDescriptionsFields: function(files, descriptions, uploadType) {
+        var filesDescriptions = $(this.element).find('.files__descriptions').first();
+        var mainDiv = null;
+        var divLabel = null;
+        var divTextarea = null;
+        var divImage = null;
+        var img = null;
+        var textarea = null;
+        var descriptionsExists = true;
+
+        this.filesDescriptions = descriptions || [];
+
+        $(filesDescriptions).show().html('');
+
+        for (var i = 0; i < files.length; i++) {
+            mainDiv = $('<div/>');
+
+            divLabel = $('<div/>');
+            divLabel.addClass('submission__file__description__label');
+            divLabel.text(gettext("Describe ") + files[i].name + ' ' + gettext("(required):"));
+            divLabel.appendTo(mainDiv);
+
+            divTextarea = $('<div/>');
+            divTextarea.addClass('submission__file__description');
+            textarea = $('<textarea />');
+            if ((this.filesDescriptions.indexOf(i) !== -1) && (this.filesDescriptions[i] !== '')) {
+                textarea.val(this.filesDescriptions[i]);
+            } else {
+                descriptionsExists = false;
+            }
+            textarea.addClass('file__description file__description__' + i);
+            textarea.appendTo(divTextarea);
+
+            if (uploadType === "image") {
+                img = $('<img/>', {
+                    src: window.URL.createObjectURL(files[i]),
+                    height: 80
+                });
+                img.onload = function() {
+                    window.URL.revokeObjectURL(this.src);
+                };
```

Comment:
Can you give `img` the default alt attribute of `Thumbnail view of <file name>`? That will satisfy [accessibility requirements](https://dequeuniversity.com/rules/axe/2.1/image-alt)